### ISSUE TITLE
Validate package versions

### ISSUE THREAD

**Issue by jacksonrayhamilton** [charlierudolph/dependency-lint#16]
Say that I have a `package.json` like the following:

``` js
{
  "dependencies": {
    "lodash": "^4.3.0"
  }
}
```

But in my `node_modules` directory, I have version `3.10.1` of `lodash` installed.

When I `require('lodash')` in a program, Node does not warn me that version of the package fails to satisfy my `semver` requirements. This can cause confusing errors, e.g. when one works on a Node.js project and switches to a branch which requires an updated version of a library, but forgets to run `npm install`.

Therefore, I think it'd be valuable if this package would check for failures to satisfy semver requirements.

**Comment by charlierudolph** [comment]
Published in 3.1.0. This was added to the previous check that all modules listed in the package.json are installed. Now it verifies that all are installed with the correct semantic version.

**Comment by jacksonrayhamilton** [comment]
Awesome, thanks!
